dss_client/test/dotnet_test_app/Program.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+ 
+using System.Runtime.InteropServices;
+
+using System.Diagnostics;
+
+using System.Security.Cryptography; // for System.Security.Cryptography.MD5
+
+unsafe public class Tester
+{
+        public const string dss_lib = "libdss.so";
+        [DllImport(dss_lib, EntryPoint="DSSClientInit")]
+        unsafe static extern void* DSSClientInit(string end_point, string access_key, string secret_key, string uuid, int name_len);
+        [DllImport(dss_lib, EntryPoint="PutObjectBuffer")]
+        unsafe static extern int PutObjectBuffer(void* dss_client, string key, int key_len, byte[] buf, long buf_size);
+
+        [DllImport(dss_lib, EntryPoint="GetObjectBuffer")]
+        unsafe static extern int GetObjectBuffer(void* dss_client, string key, int key_len, byte* buf, long buf_size);
+
+        [DllImport(dss_lib, EntryPoint="GetObject")]
+        static extern int GetObject(void* dss_client, string key, int key_len, string dest_fn);
+        [DllImport(dss_lib, EntryPoint="PutObject")]
+        static extern int PutObject(void* dss_client, string key, int key_len, string src_fn);
+
+        [DllImport(dss_lib, EntryPoint="DeleteObject")]
+        static extern int DeleteObject(void* dss_client, string key, int key_len);
+        [DllImport(dss_lib, EntryPoint="ListObjects")]
+        unsafe static extern IntPtr ListObjects(void* dss_client, string prefix, string delimit);
+        static public string CalculateMD5(string filename)
+        {
+                using (var md5 = MD5.Create())
+                {
+                        using (var stream = File.OpenRead(filename))
+                        {
+                                var hash = md5.ComputeHash(stream);
+                                return BitConverter.ToString(hash).Replace("-", "").ToLowerInvariant();
```

Comment:
do we need to do file close or the file description gets automatically released?

Reply:
I think It is automatically closed at the end of the "using" scope.